Java-Loops.md

```diff
@@ -0,0 +1,208 @@
+# Java Loops
+
+Whenever you need to execute a block of code multiple times, a loop will often come in handy. In general, a loop looks something like this:
+
+![Loop Schematic Diagram](http://cse.iitkgp.ac.in/pds/notes/img/loop2.gif)
+
+## Types of Loops
+
+Java has 4 types of loops - the `while`, `do while`, `for` and `for each`. While you can pick your favorite and use that almost everywhere, each of them have certain advantages in different applications.
+
+### The While loop
+
+The `while` statement evaluates the `expression` (which must return a boolean value), and if it is `true`, the `Statements` get executed.
+
+```java
+while (expression) 
+{
+  // Statements
+}
+```
+
+In the following example, the `expression` is given by `iter_While < 10`. As we increment `iter_While` by `1` each time the loop is executed, the `while` loop will keep going till `iter_While` reaches `10`.
+
+```java
+int iter_While = 0;
+while (iter_While < 10) 
+{
+    System.out.print (iter_While + " ");
+    // Increment the counter
+    // Iterated 10 times, iter_While 0,1,2...9
+    iter_While++;
+}
+System.out.println ("iter_While Value: " + iter_While);
+```
+
+Output:
+
+    0 1 2 3 4 5 6 7 8 9
+    iter_While Value: 10
+
```

Comment:
A good point would be to mention here that one can use Eclipse or IntelliJ or Netbeans to debug it with breakpoints. Breakpoints would help visualize how the code works step-by-step. No need to explain how to set breakpoints etc., just link to some articles or blog posts.

This is good, because this sets up a Java developer from the very beginning to use debugger and other IDE tools.